raft/src/test/java/org/apache/kafka/raft/QuorumStateTest.java

```diff
@@ -154,6 +155,163 @@ private ReplicaKey replicaKey(int id, boolean withDirectoryId) {
         return ReplicaKey.of(id, directoryId);
     }
 
+    @ParameterizedTest
```

Comment:
I've organized QuorumStateTest in the following way - misc tests were pulled to the front. All other tests are organized under banners (e.g. Initialization tests, Tests of transitions from state X)

Reply:
We should have done this in a different PR. It is difficult for me to see what has change and what has move so I have to review almost the entire file.

Reply:
Sorry, I wish github was a bit smarter with diffs :( It was difficult for me to figure out what coverage we were missing without the tests being more ordered (and I thought it would be difficult for you to tell what we might be missing as well) so I ended up deciding the re-order was worth it. 
We discussed this briefly before, but ideally each state will have its own file in the end - I decided not to make that change in this PR since it would make it even harder to tell what had changed.